crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -625,15 +625,32 @@ where
                     orelse,
                 },
             ) => {
-                // TODO add control flow similar to `ast::Stmt::While` above
                 self.add_standalone_expression(iter);
                 self.visit_expr(iter);
+
+                let pre_loop = self.flow_snapshot();
+                let saved_break_states = std::mem::take(&mut self.loop_break_states);
+
                 debug_assert!(self.current_assignment.is_none());
                 self.current_assignment = Some(for_stmt.into());
                 self.visit_expr(target);
                 self.current_assignment = None;
+
```

Comment:
We should have a TODO here for the actual loopy part of flow control: the fact that definitions in the loop body... are visible to the loop body. But we don't want to implement this until we have fixpoint iteration support in Salsa, or else we'll just panic a lot with cycle problems.

I think I also should have added such a TODO to the `while` control flow, and didn't. But I guess that could also be added in this PR?